SPECS/sudo/sudo.spec

```diff
@@ -15,6 +15,7 @@ BuildRequires:  openssl-libs
 BuildRequires:  pam-devel
 BuildRequires:  sed
 BuildRequires:  zlib-devel
+BuildRequires:  openldap
```

Comment:
The documentation suggests `openldap-devel`: https://www.sudo.ws/docs/install/#linux

Reply:
Looking at sudo build options when enabling ldap, it requires access to iber.h and ldap.h, I took a look at the mariner-supplied packages, both these headerfiles are provided in openldap

```
azaugg@azaugg-ld77 [ ~/CBL-Mariner/SPECS/sudo ]$ sudo repoquery -l openldap|grep inc
Last metadata expiration check: 0:01:52 ago on Thu May 18 15:26:41 2023.
- Fix CVE-2022-29155
/usr/include/lber.h
/usr/include/lber_types.h
/usr/include/ldap.h
/usr/include/ldap_cdefs.h
/usr/include/ldap_features.h
/usr/include/ldap_schema.h
/usr/include/ldap_utf8.h
/usr/include/ldif.h
/usr/include/openldap.h
/usr/include/slapi-plugin.h
```

Reply:
Hmm, come to think of it, I don't think Mariner ships with a devel package for openldap
```
azaugg@azaugg-ld77 [ ~/CBL-Mariner/SPECS/sudo ]$ sudo yum info openldap-devel
[sudo] password for azaugg:
Loaded plugin: tdnfrepogpgcheck
[using capability match for 'openldap-devel']
Name          : openldap
Arch          : x86_64
Epoch         : 0
Version       : 2.4.57
Release       : 8.cm2
Install Size  :   1.65M 1732185 (1732185)
Repo          : @System
...
...
...
```

Reply:
Hmm you are right. Our [openldap spec file](https://github.com/microsoft/CBL-Mariner/blob/main/SPECS/openldap/openldap.spec) doesn't specify a devel package. Let me check back on this

Reply:
I would still use the `openldap-devel` name that's currently attached to the main package. It makes less work for us when we inevitably split the package properly in a future release.

Reply:
Done, referenced to openldap-devel. I do want to point out that if a devel package is needed, we must have the includes iber.h and ldap.h in the devel file